library/cohesity_restore_file.py

```diff
@@ -401,6 +437,40 @@ def start_restore(module, uri, self):
         raise__cohesity_exception__handler(error, module)
 
 
+def get__job_information__for_file(module, source_object_info):
+    try:
+        # Based on job id fetch run details based on used provided timestamp.
+        # If timestamp is not provided fetch latest successful run details.
+        run_id = None
+        job_runs = cohesity_client.protection_runs.get_protection_runs(
+            job_id=source_object_info["jobId"])
+        for run in job_runs:
+            if not module.params.get('backup_timestamp') and run.backup_run.status == "kSuccess":
+                run_id = run.backup_run.job_run_id
+                t_secs = run.backup_run.stats.start_time_usecs
+                break
+            else:
+                snapshot_timestamp = datetime.strptime(
```

Comment:
This will search search backup run in unsuccessfull runs ? why?

Reply:
public/ProtectionRuns will returns all the available runs for a job id. If used proivded backup timestamp we check for the timestamp and restore file at that point or else latest  snapshot is used.